org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinComposeFilterTest.java

```diff
@@ -253,4 +253,107 @@ public void should_filter() {
 				range7);
 	}
 
+	/**
+	 * <pre>
+	 * &#064;androidx.compose.runtime.Composable
+	 * fun example() {
+	 *   ... // body
+	 * }
+	 * </pre>
+	 *
+	 * after <a href=
+	 * "https://github.com/JetBrains/kotlin/commit/ee9217f8f0f37967684fbfe4a568c2b3c8707507">
+	 * change in Compose Kotlin compiler plugin version 2.1.0</a> transformed
+	 * into
+	 *
+	 * <pre>
+	 * fun example($composer: Composer?, $changed: Int) {
+	 *   $composer = $composer.startRestartGroup(...)
+	 *   sourceInformation($composer, ...)
+	 *   if ($composer.shouldExecute(...)) {
+	 *     ... // body
+	 *   } else {
+	 *     $composer.skipToGroupEnd()
+	 *   }
+	 *   $composer.endRestartGroup()?.updateScope { ... }
+	 * }
+	 * </pre>
+	 *
+	 * This <a href=
+	 * "https://github.com/JetBrains/kotlin/commit/5f7e5d1518e839c1f8514a5c145aad52b1ce1739">became
+	 * the default in Compose Kotlin compiler plugin version 2.2.0</a>
+	 */
+	@Test
+	public void should_filter_pausable_composition() {
+		final MethodNode m = new MethodNode(InstrSupport.ASM_API_VERSION, 0,
+				"example", "(Landroidx/compose/runtime/Composer;I)V", null,
+				null);
+		m.visitAnnotation("Landroidx/compose/runtime/Composable;", false);
+
+		final Range range1 = new Range();
+		final Range range2 = new Range();
+
+		m.visitVarInsn(ALOAD, 0);
+		range1.fromInclusive = m.instructions.getLast();
+		m.visitLdcInsn(Integer.valueOf(-933543558));
```

Comment:
Just out of curiosity: What is this specific number about?

Reply:
@marchof "FunctionSourceKey" which is some hash code of function - see
* https://github.com/JetBrains/kotlin/blob/c448af19ded1b1a4e96e9af6412cd9acb100ce1a/plugins/compose/compiler-hosted/src/main/java/androidx/compose/compiler/plugins/kotlin/lower/ComposableFunctionBodyTransformer.kt#L1200-L1204
* https://github.com/JetBrains/kotlin/blob/c448af19ded1b1a4e96e9af6412cd9acb100ce1a/plugins/compose/compiler-hosted/src/main/java/androidx/compose/compiler/plugins/kotlin/lower/ComposableFunctionBodyTransformer.kt#L1953-L1956